finality-aleph/src/network/service.rs

```diff
@@ -112,13 +116,13 @@ impl<N: Network, D: Data> Service<N, D> {
                         match network.sender(peer_id, protocol) {
                             Ok(sender) => senders.entry(protocol).or_insert(sender),
                             Err(e) => {
-                                debug!(target: "aleph-network", "Failed creating sender. Dropping message: {:?}", e);
+                                info!(target: "aleph-network", "Failed creating sender. Dropping message: {}", e);
```

Comment:
Are we sure this will not create spam? I definitely want to avoid a situation where the log is spammed with this. 
Each type of message should be upper bounded to appear once per 10s or some such. 

Similarly for `Failed sending data to peer`.

Just create some aggregate statistics on failures and print them every 10 sec.

Reply:
After some research this will only log in case `StreamClosed` notification did not happen before trying to send here... or if there is some strange bug in substrate. So seems like this log might be undeterministic ://  Also after some thinking this does not sound like info log. Aggregation could be the way to go.

Reply:
There is one problem though. This function is called per peer, so this is not that easy to aggregate statistics from all those at one place. I don't think sending such data over channels is worth it. That said we could add this aggregation to `send_to_peer` function. Everything that is sent to senders goes through this function. I think there we can aggregate each failure to send.

Still, I'm wondering, what exactly should we aggregate:
*  Should we aggregate number of unsent messages per peer? (I guess there might be a lot of peers missing, and this could cause a lot of spam) 
* Should we aggregate just what peers we failed to send? (same nit against as in above point)
* Just aggregating number of unsent messages is not that useful. 

And also for how long should we aggregate them? Should we print peers/messages lost since last status report? I think something like this would be the way to go, but 10 seconds seems like a really short time to reset such aggregation.

So to sum up, I would personally do this aggregation in `send_to_peer`, aggregate unsent messages per user and reset it once every 5-10 status reports?

Reply:
For context, I suggested this change because there were _no_ such logs (at the appropriate level) even when there were problems with connections – as far as I can tell, these are really rare. Sending failures usually turn up in a completely different part of the code.

On the other hand maybe this is not good enough reasoning for level bumping, so perhaps we shouldn't be doing that. I'm kinda skeptical the setup needed to aggregate failure logs here is worth it, especially since, as @maciejnems explained it would have to add some centralization and cross-thread communication, which is usually more complex than one would hope for in logging code.

Reply:
I cannot really provide any useful insight here, on top of what @timorl already said. Aggregation per peer does not make much sense to me, as it does not guarantee satisfaction of the no-spam condition. 

If the only satisfying solution is complex, maybe you should just move it out of scope of this PR, and create a new task for it?

Reply:
Oh yeah, this specific change will become mostly irrelevant with the separate validator network, duh. >.<

Reply:
Ok, then if you think this will not create much spam, you can leave it on `info`, otherwise better to change to `debug` and leave it for the future.

Reply:
I changed it back to debug